sw/host/tpm2_test_server/src/main.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::interface::serve_command;
+use log::LevelFilter;
+use mio::{Events, Interest, Poll, Token};
+use opentitanlib::backend;
+use opentitanlib::io::i2c::I2cParams;
+use opentitanlib::io::spi::SpiParams;
+use opentitanlib::tpm::{Driver, I2cDriver, SpiDriver};
+use std::net::{SocketAddr, TcpListener};
+use structopt::StructOpt;
+
+mod interface;
+
+#[derive(Debug, StructOpt)]
+pub enum TpmBus {
+    Spi {
+        #[structopt(flatten)]
+        params: SpiParams,
+    },
+    I2C {
+        #[structopt(flatten)]
+        params: I2cParams,
+    },
+}
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "tpm2-test-server",
+    about = "A tool for processing TPM commands over a TCP port."
+)]
+struct Opts {
+    #[structopt(subcommand)]
+    bus: TpmBus,
+
+    #[structopt(long, default_value = "off")]
+    logging: LevelFilter,
+
+    #[structopt(flatten)]
+    backend_opts: backend::BackendOpts,
+
+    #[structopt(
+        short,
+        long,
+        default_value = "9883",
+        help = "TCP port for incoming connections."
+    )]
+    tpm_port: u16,
+}
+
+const CMD_TOKEN: Token = Token(0);
+const PLATFORM_TOKEN: Token = Token(1);
+
+pub fn main() -> std::io::Result<()> {
```

Comment:
What program will be the client to `tpm2_test_server`?

Reply:
Either [Tpm2Tester](https://github.com/TrustedComputingGroup/Compliance_PC-TPM-Internal), a sample test suite, or [Comliance_PC-TPM-Internal](https://github.com/TrustedComputingGroup/Compliance_PC-TPM-Internal), a test suite from TCG available to members.